bump version to 1.4

benvial opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

Commit:
bumb version to 1.4

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

benvial commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 3.21.9, conda-smithy 3.21.0, and conda-forge-pinning 2022.06.29.00.57.15

benvial commented:
@ericmjl @richardotis , I have updated the recipe according to the pull request [#8](https://github.com/conda-forge/autograd-feedstock/pull/8):

- added the license file
- set the minimum python version to 3.6 (according to [this](https://github.com/HIPS/autograd/pull/368) versions <3.6 are not supported). That means dropping Python 2.7.

Feel free to merge, it would be great to have the latest autograd on conda-forge.

richardotis reviewed (approved):
Thanks for putting in the work to update this.
